### PR TITLE
add linkerd example with header based routing

### PR DESCRIPTION
**Pull request by AlonGluz**
https://github.com/argoproj-labs/rollouts-plugin-trafficrouter-gatewayapi/issues/84

### CODE REVIEW

**Comment by AlonGluz** [comment]
@kostis-codefresh Anything wrong with this PR?

File: examples/linkerd-header-based/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+```shell
+kubectl apply -f rollout.yaml
+```
+
+## Step 9 - Watch the rollout
+```shell
+watch "kubectl -n default get httproute.gateway.networking.k8s.io/argo-rollouts-http-route -o custom-columns=NAME:.metadata.name,PRIMARY_SERVICE:.spec.rules[0].backendRefs[0].name,PRIMARY_WEIGHT:.spec.rules[0].backendRefs[0].weight,CANARY_SERVICE:.spec.rules[0].backendRefs[1].name,CANARY_WEIGHT:.spec.rules[0].backendRefs[1].weight"
+```
+
+## Step 10 - Patch the rollout to see the canary deployment
```
Comment (kostis-codefresh) [inline examples/linkerd-header-based/README.md:87]: This is great. But in this example we also want to test header based routing specifically. How do we know that it works?

Can we add 2 examples of "curl" commands here, one with the header and one without the header to see what happens?

**Review by kostis-codefresh — CHANGES_REQUESTED** [review]
I think we need some extra instructions in the README specifically for header based routing.

**Comment by AlonGluz** [comment]
@kostis-codefresh sure, I'll take care of it 👍🏻

**Comment by AlonGluz** [comment]
@kostis-codefresh I've applied the PR comments and tested it with L5d version 2.14.10, LMK if there are any other blockers

**Pull request merged**